rapid_response_xblock/static/js/src/rapid.js

```diff
@@ -43,6 +46,227 @@
+        return {
+          answer_id: response.answer_id,
+          answer_text: response.answer_text,
+          count: lookup[response.answer_id]
+        };
+      });
+    }
+
+    function makeIntegerTicks(domain) {
+      // Given the domain limits return 6 or so tick values, equally spaced out, all integers.
+      var tickValues = [];
+      var lookup = {};
+      var maxCount = domain.domain()[1];
+      if (!isNaN(maxCount)) {
+        var numTicks = 6;
+        var tickIncrement = Math.ceil(maxCount / numTicks);
+        for (var tickCount = 0; tickCount < numTicks; ++tickCount) {
+          var tick = tickCount * tickIncrement;
+          if (!(tick in lookup)) {
+            lookup[tick] = true;
+            tickValues.push(tick);
+          }
+        }
+      }
+      return tickValues;
+    }
+
+    function wrapText(textSelector, barWidth, oldText) {
+      // see https://bl.ocks.org/mbostock/7555321 for inspiration
+      // SVG doesn't have a capability to wrap text except for foreignObject which is not supported in IE11.
+      // So we have to calculate it manually
+      textSelector.each(function() {
+        var root = d3.select(this);
+
+        var rootY = root.attr("y");
+        var rootDy = parseFloat(root.attr("dy"));
+
+        root.selectAll("tspan").remove();
+        var words = oldText.split(/\s+/);
+        var tspan = root.append("tspan").attr("x", 0).attr("y", rootY).attr("dy", rootDy + "em");
+
+        var currentLine = 0;
+        var lineHeight = 1.1;
+        words.forEach(function(word) {
+          if (!word) {
+            return;
+          }
+
+          var oldText = tspan.text();
+          tspan.text(oldText + " " + word);
+          if (tspan.node().getComputedTextLength() > barWidth) {
+            currentLine++;
+            tspan.text(oldText + " ");
+            tspan = root.append("tspan").attr("x", 0).attr("y", rootY).attr(
+              "dy", ((currentLine * lineHeight) + rootDy) + "em"
+            ).text(word);
+          }
+        });
+      });
+    }
+
+    function renderD3() {
+      // This function should be called whenever the response information changes.
+
+      var message = chart.select(".message");
+      var responses = state.responses;
+      if (responses.length === 0) {
+        message.text("No data available").classed("hidden", false);
+      } else {
+        message.classed("hidden", true);
+      }
+
+      // Compute responses into information suitable for a bar graph.
+      var histogram = makeHistogram(responses);
+      var histogramLookup = {};
+      histogram.forEach(function(response) {
+        histogramLookup[response.answer_id] = response;
+      });
+
+      // Add answer ids to the color domain if they don't already exist
+      histogram.forEach(function(response) {
+        var answerId = response.answer_id;
+        if (!_.includes(colorDomain, answerId)) {
+          colorDomain.push(answerId);
+        }
+      });
```

Comment:
I think this can be rewritten as:

```javascript
colorDomain = _.union(colorDomain, _.pluck(histogram, 'answer_id'));
```

Obviously the pluck wouldn't be needed if we stored the answer_id array separately (see comment below)